open_feature/open_feature_client.py

```diff
@@ -271,12 +325,30 @@ def _create_provider_evaluation(
 
         get_details_callable = {
             FlagType.BOOLEAN: self.provider.get_boolean_details,
-            FlagType.NUMBER: self.provider.get_number_details,
+            FlagType.INTEGER: self.provider.get_integer_details,
+            FlagType.FLOAT: self.provider.get_float_details,
             FlagType.OBJECT: self.provider.get_object_details,
             FlagType.STRING: self.provider.get_string_details,
         }.get(flag_type)
 
         if not get_details_callable:
             raise GeneralError(error_message="Unknown flag type")
 
-        return get_details_callable(*args)
+        value = get_details_callable(*args)
+
+        if flag_type in NUMERIC_TYPES:
```

Comment:
I don't think we need to do this anymore, right? We can just rely on the provider providing the correct type from their methods?

Reply:
Added an `isinstance` check instead